src/ui/public/vislib/visualizations/line_chart.js

```diff
@@ -291,6 +291,9 @@ export default function LineChartFactory(Private) {
 
         var layers = data.series.map(function mapSeries(d) {
           var label = d.label;
+          if (self._attr.hideZeros) {
+            d.values = d.values.filter(function (e) { return e.y; });
+          }
```

Comment:
This functionality should be moved [here](https://github.com/worldline/kibana/blob/zero-values/src/ui/public/vislib/visualizations/line_chart.js#L196).

The problem with the way it is done here is that it will connect the lines between the points where there are zero values. The zero values actually represent nulls, which means that there should be gaps between the points connected to the null (or zero value).

I am still confused here though. If the original data contains nulls, and somewhere we are summing nulls to zero or if the actual data has zeros.